src/Services/Air/AirParser.js

```diff
@@ -1082,8 +1082,16 @@ function exchangeBooking(rsp) {
 
 function availability(rsp) {
   const itinerarySolution = utils.firstInObj(rsp['air:AirItinerarySolution']);
-  const connectedSegments = itinerarySolution['air:Connection']
-    ? itinerarySolution['air:Connection'].map(
+
+  // Sometimes might return an object instead of array
+  let airConnections = itinerarySolution['air:Connection'];
+
+  if (airConnections && !airConnections.length && Object.keys(airConnections).length !== 0) {
+    airConnections = [airConnections];
```

Comment:
Please try configuring `uapi-parser` by setting non-collapsable field [here](https://github.com/Travelport-Ukraine/uapi-json/blob/master/src/Request/default-config.js#L7) instead of manually checking for length